src/views/Home/components/CLWCreditsDaysLeft/index.js

```diff
@@ -11,194 +11,192 @@ import './CLWChart.css';
 
 import { Card, CardHeader, CardContent, Typography, Grid, Button } from '@material-ui/core';
 
-export default class CLWCreditsDaysLeft extends Component {
-  constructor(props) {
-    super(props);
-
-    this.loadData = this.loadData.bind(this);
-
-    this.state = {
-      daysLeft: [],
-      chapelCredits: {},
-      error: null,
-      loading: true,
-      currSessionDescription: '',
+const CLWCreditsDaysLeft = () => {
+  const [state, setState] = useState({
+    daysLeft: [],
+    chapelCredits: {},
+    error: null,
+    loading: true,
+    currSessionDescription: '',
+  });
+
+  useEffect(() => {
+    const loadData = async () => {
+      setState({ loading: true });
+      try {
+        const daysLeftPromise = session.getDaysLeft();
+        const chapelCreditsPromise = user.getChapelCredits();
+        const currSessionPromise = session.getCurrent();
+        const daysLeft = await daysLeftPromise;
+        const chapelCredits = await chapelCreditsPromise;
+        const currSession = await currSessionPromise;
+        const currSessionDescription = currSession.SessionDescription.replace(
+          /(Academic Year)|(Grad)/gm,
+          '',
+        );
+        setState({
+          loading: false,
+          daysLeft: daysLeft,
+          chapelCredits: chapelCredits,
+          currSessionDescription: currSessionDescription,
+        });
+      } catch (error) {
+        setState({ error });
+      }
     };
-  }
-
-  componentDidMount() {
-    this.loadData();
-  }
+    loadData();
+  }, []);
 
-  async loadData() {
-    this.setState({ loading: true });
-    try {
-      const daysLeftPromise = session.getDaysLeft();
-      const chapelCreditsPromise = user.getChapelCredits();
-      const currSessionPromise = session.getCurrent();
-      const daysLeft = await daysLeftPromise;
-      const chapelCredits = await chapelCreditsPromise;
-      const currSession = await currSessionPromise;
-      const currSessionDescription = currSession.SessionDescription.replace(
-        /(Academic Year)|(Grad)/gm,
-        '',
-      );
-      this.setState({ loading: false, daysLeft, chapelCredits, currSessionDescription });
-    } catch (error) {
-      this.setState({ error });
-    }
+  if (state.error) {
+    throw state.error;
```

Comment:
This is an anti-pattern that we should try to remove from the code. The only reason to catch an error is to handle it in a particular way. Instead, what we're doing is catching an error and setting it as part of the state, then checking state and re-throwing the error. That's actually worse than never catching the error at all. IF you don't catch the error, it will continue to propagate up the call stack until it gets caught elsewhere. However, since we're catching it, storing it in state, and then rethrowing it, we have gotten rid of the call stack info showing where exactly the error came from.